guide/message_transformation/message_transformation.bal

```diff
@@ -0,0 +1,370 @@
+                table dt1;
+                match selectRet1 {
+                    table tableReturned => dt1 = tableReturned;
+                    error e => io:println("Select data from StudentDetails table failed: "
+                            + e.message);
+                }
+                //convert the details to a jason file
+                io:println("\nConvert the table into json");
+                var jsonConversionRet1 = <json>dt1;
+                match jsonConversionRet1 {
+                    json jsonRes1 => {
+                        //set student's result details to the global variable
+                        payload2= untaint jsonRes1;
+                    }
+                    error e => io:println("Error in StudentDetails table to json conversion");
+                }
+                testDB.stop();
+
+                //JSON to JSON conversion to the selected details
+                //define new jason variable
+                json pay = payload1[0];
+                //add extra values to the jason payload
+                pay.fname = pay.name;
+                //remove values from the jason payload
+                pay.remove("name");
+                //add results to the same payload
+                pay.results = payload2[0];
+                //set enriched payload to the request
+                enrichedreq.setJsonPayload(pay);
+            }
+            error err => {
+                http:Response res = new;
+                res.statusCode = 500;
+                res.setPayload(untaint err.message);
+                caller->respond(res) but { error e =>
+                log:printError("Error sending response", err = e) };
+            }
+        }
+        //forward enriched request to the client endpoint
+        var clientResponse = clientEP->forward("/", enrichedreq);
+        match clientResponse {
+            http:Response res => {
+                caller->respond(res) but { error e =>
+                log:printError("Error sending response", err = e) };
+            }
+            error err => {
+                http:Response res = new;
+                res.statusCode = 500;
+                res.setPayload(err.message);
+                caller->respond(res) but { error e =>
+                log:printError("Error sending response", err = e) };
+            }
+        }
+    }
+}
+
+//client endpoint service to display the request payload
+service<http:Service> backend bind backendEP {
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/"
+    }
+    backendservice (endpoint caller, http:Request enrichedreq) {
+        //get the requset payload
+        var jsonMsg = enrichedreq.getJsonPayload();
+        match jsonMsg {
+            json msg => {
+                //send payload as response
+                http:Response res = new;
+                res.setJsonPayload(untaint msg);
+                caller->respond(res) but { error e =>
```

Comment:
We can directly send the json message via respond call like this "caller -> respond(untaint jsonMsg)" instead of creating a response object since you only set the payload